bonita-migration-distrib/src/it/groovy/org/bonitasoft/migration/DBUnitHelper.groovy

```diff
@@ -0,0 +1,165 @@
+    }
+
+    def static String[] createTables(sql, String version, String feature) {
+        println "Create tables of $feature"
+        getCreateTables(version, feature).text.split("@@").each({ stmt ->
+            sql.execute(stmt)
+        })
+    }
+
+    def static boolean hasTable(Sql sql, String tableName) {
+        def firstRow
+        switch (dbVendor()) {
+            case POSTGRES:
+                def query = """
+                    SELECT *
+                     FROM information_schema.tables
+                     WHERE table_schema='public'
+                       AND table_type='BASE TABLE'
+                       AND table_name = ?
+                    """
+                firstRow = sql.firstRow(query, tableName)
+                break
+
+            case ORACLE:
+                def query = """
+                    SELECT *
+                    FROM user_tables
+                    WHERE table_name = ?
+                    OR table_name = ?
+                    """ as String
+                firstRow = sql.firstRow(query, [tableName, tableName.toUpperCase()])
+                break
+
+            case MYSQL:
+                def query = """
+                    SELECT *
+                    FROM information_schema.tables
+                    WHERE table_name = ?
+                    AND table_schema = DATABASE()
+                    """
+                firstRow = sql.firstRow(query, tableName)
+                break
+
+            case "sqlserver":
+                def query = """
+                    SELECT * FROM information_schema.tables
+                    WHERE TABLE_NAME = ?
+                    """
+                firstRow = sql.firstRow(query, tableName)
+                break
+        }
+        return firstRow != null
+    }
+
+    def static JdbcDatabaseTester createTester() {
+        new JdbcDatabaseTester(driverClass, url, user, password) {
+            public IDatabaseConnection getConnection() {
+                if (dbVendor() == ORACLE) {
+                    def conn = DriverManager.getConnection(getUrl(), getUser(), getPassword());
+                    return new OracleConnection(conn, getUser());
+                } else {
+                    return super.getConnection();
+                }
+            }
+        }
+    }
+
+    def static Sql createSqlConnection() {
+        Sql.newInstance(url, user, password, driverClass)
+    }
+
+    def static String getPassword() {
+        System.getProperty("dbpassword")
+    }
+
+    def static String getUser() {
+        System.getProperty("dbuser")
+    }
+
+    def static String getUrl() {
+        System.getProperty("dburl")
+    }
+
+    def static String getDriverClass() {
+        System.getProperty("dbdriverClass")
+    }
+
+    def static dropTables(Sql sql, String[] tables) {
+        tables.each {
+            sql.execute("DROP TABLE $it".toString())
```

Comment:
Actually, this is a workaround for a replacement issue: the parameter is not correctly replaced without ".toString". I'll add a note.